.github/workflows/lint.yml

```
@@ -14,7 +14,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
+      - uses: actions/setup-python@v5.1.0
```

Should we use hashes here too?
```suggestion
      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
```

---

I don't think it's necessary, especially as it's only reading code and not producing any artifacts.